### PR TITLE
Better scrolling and scroll state

### PR DESCRIPTION
**Pull request by gregorylegarec**
This PR removes the complexity of ScrollToTop component and let the browser handle the scrolling of different elements.

The main trick is done by always rendering the app list and treat ApplicationPage as a modal page, positionned on top.

There is a few difference between desktop and mobile (as position `absolute` or `fixed`) but the whole idea seems to work.

Edit (CPatchane) Closes #412

### CODE REVIEW

File: src/ducks/apps/components/MyApplications.jsx [hunk 1]
```diff
@@ -48,23 +48,21 @@ export class MyApplications extends Component {
     return (
       <ScrollToTop target={this.container}>
         <Content className="sto-myapps" ref={div => (this.container = div)}>
-          {this.props.match.isExact ? (
```
Comment (gregorylegarec) [inline src/ducks/apps/components/MyApplications.jsx]: Always render

File: src/ducks/apps/components/Discover.jsx [hunk 2]
```diff
@@ -51,24 +51,22 @@ export class Discover extends Component {
     return (
       <ScrollToTop target={this.container}>
         <Content className="sto-discover" ref={div => (this.container = div)}>
-          {this.props.match.isExact ? (
```
Comment (gregorylegarec) [inline src/ducks/apps/components/Discover.jsx]: Always render

File: src/styles/application.css [hunk 3]
```diff
@@ -1,5 +1,13 @@
 .sto-app {
   margin: auto 1.7em;
+  height: 100vh;
```
Comment (GoOz) [inline src/styles/application.css]: `100vh` is a risky move to do, especially on mobile. 
For mobile browsers, 100vh > 100%

File: src/styles/application.css [hunk 4]
```diff
@@ -1,5 +1,13 @@
...
+  background-color: white;
+}
+
+@media (max-width: 48rem) {
+  .sto-app {
+    position: inherit;
```
Comment (GoOz) [inline src/styles/application.css]: `inherit` from what ? I don't get what it's for.

**Comment by gregorylegarec** [comment]
@CPatchane Back button alignment fixed in https://github.com/cozy/cozy-store/issues/422

**Comment by gregorylegarec** [comment]
@CPatchane Back button alignment fixed in https://github.com/cozy/cozy-store/issues/422

**Comment by gregorylegarec** [comment]
@CPatchane Back button alignment fixed in https://github.com/cozy/cozy-store/issues/422

File: src/styles/application.css (hunk 3 above)
Comment (gregorylegarec) [reply src/styles/application.css]: Fixed in the following media query rule.

**Pull request merged**